crates/matrix-sdk-common/src/linked_chunk/mod.rs

```diff
@@ -1474,11 +1483,27 @@ mod tests {
         assert!(LinkedChunk::<3, char, ()>::new_with_update_history().updates().is_some());
     }
 
+    #[test]
+    fn test_new_with_initial_update() {
+        use super::Update::*;
+
+        let mut linked_chunk = LinkedChunk::<3, char, ()>::new_with_update_history();
+
+        assert_eq!(
+            linked_chunk.updates().unwrap().take(),
+            &[NewItemsChunk { previous: None, new: ChunkIdentifier(0), next: None }]
+        );
+    }
+
     #[test]
     fn test_push_items() {
         use super::Update::*;
 
         let mut linked_chunk = LinkedChunk::<3, char, ()>::new_with_update_history();
+
+        // Ignore initial update.
+        let _ = linked_chunk.updates().unwrap().take();
```

Comment:
Maybe we could also check (here and everywhere below) that the initial update is what we expected?

Reply:
That's the `test_new_with_initial_update` test, just above this one.